metadata/resources_repository_test.go

```diff
@@ -0,0 +1,767 @@
+			Features: NameVariants{
+				{"feature", "variant"},
+			},
+			Owner:      "Other",
+			Tags:       Tags{},
+			Properties: Properties{},
+		},
+	}
+
+	// Initialize the test server once for all subtests
+	ctx := context.Background()
+	testServer := newTestMetadataServer(t)
+	defer testServer.Close()
+
+	resourceIds := make([]ResourceID, 0)
+	for _, res := range resources {
+		resourceIds = append(resourceIds, res.ResourceID())
+	}
+
+	testServer.ResetDatabase()               // Reset DB before test
+	testServer.SeedResources(ctx, resources) // Seed data
+	testServer.SetResourcesReady(ctx, resourceIds)
+
+	t.Run("Attempt to delete resource with dependencies", func(t *testing.T) {
+		err := testServer.repo.MarkForDeletion(ctx, common.ResourceID{
+			Name:    "mockSource",
+			Variant: "var",
+			Type:    common.SOURCE_VARIANT,
+		}, testServer.server.deletionTaskStarter)
+		require.Error(t, err)
+	})
+
+	t.Run("Delete training set", func(t *testing.T) {
+		err := testServer.repo.MarkForDeletion(ctx, common.ResourceID{
+			Name:    "training-set",
+			Variant: "variant",
+			Type:    common.TRAINING_SET_VARIANT,
+		}, testServer.server.deletionTaskStarter)
+		require.NoError(t, err)
+
+		res, err := testServer.repo.Lookup(ctx, ResourceID{
+			Name:    "training-set",
+			Variant: "variant",
+			Type:    TRAINING_SET_VARIANT,
+		}, DeleteLookupOption{DeletedOnly})
+		require.NoError(t, err)
+		require.NotNil(t, res)
+		require.Equal(t, res.ID().Name, "training-set")
+	})
+
+	t.Run("Delete feature", func(t *testing.T) {
+		err := testServer.repo.MarkForDeletion(ctx, common.ResourceID{
+			Name:    "feature",
+			Variant: "variant",
+			Type:    common.FEATURE_VARIANT,
+		}, testServer.server.deletionTaskStarter)
+		require.NoError(t, err)
+
+		res, err := testServer.repo.Lookup(ctx, ResourceID{
+			Name:    "feature",
+			Variant: "variant",
+			Type:    FEATURE_VARIANT,
+		}, DeleteLookupOption{DeletedOnly})
+		require.NoError(t, err)
+		require.NotNil(t, res)
+		require.Equal(t, res.ID().Name, "feature")
+	})
+
+	t.Run("Delete source (should fail since has deps)", func(t *testing.T) {
+		err := testServer.repo.MarkForDeletion(ctx, common.ResourceID{
+			Name:    "mockSource",
+			Variant: "var",
+			Type:    common.SOURCE_VARIANT,
+		}, testServer.server.deletionTaskStarter)
+		require.Error(t, err)
+	})
+
+	t.Run("Delete label", func(t *testing.T) {
+		err := testServer.repo.MarkForDeletion(ctx, common.ResourceID{
+			Name:    "label",
+			Variant: "variant",
+			Type:    common.LABEL_VARIANT,
+		}, testServer.server.deletionTaskStarter)
+		require.NoError(t, err)
+
+		res, err := testServer.repo.Lookup(ctx, ResourceID{
+			Name:    "label",
+			Variant: "variant",
+			Type:    LABEL_VARIANT,
+		}, DeleteLookupOption{DeletedOnly})
+		require.NoError(t, err)
+		require.NotNil(t, res)
+		require.Equal(t, res.ID().Name, "label")
+	})
+
+	t.Run("Delete source", func(t *testing.T) {
+		err := testServer.repo.MarkForDeletion(ctx, common.ResourceID{
```

Comment:
This should prob be a fn, I feel like this code has been copy and pasted atleast a dozen times